src/utils/lifecycle.js

```diff
@@ -24,11 +24,16 @@ export function hasMethod(name) {
  * @returns {Promise}
  */
 export function raceTo(promise, ms, msg) {
+  let timeout;
+
   return Promise.race([
     promise,
-    new Promise(resolve => setTimeout(() => resolve(PROMISE_TIMEOUT), ms)),
+    new Promise((resolve) => {
+      timeout = setTimeout(() => resolve(PROMISE_TIMEOUT), ms);
+    }),
   ]).then((res) => {
     if (res === PROMISE_TIMEOUT) logger.info(msg, { timeout: ms });
+    if (timeout) clearTimeout(timeout);
```

Comment:
indeed, `clearTimeout(timeout)` will always be safe, it's just slightly cleaner to use the `if` :-)